MLIR.cmake

```diff
@@ -44,14 +44,28 @@ set(
 )
 include_directories(${MLIR_INCLUDE_PATHS})
 
+# Force BUILD_SHARED_LIBS to be the same as LLVM build
+file(STRINGS ${LLVM_PROJ_BUILD}/CMakeCache.txt shared REGEX BUILD_SHARED_LIBS)
+string(REGEX REPLACE "BUILD_SHARED_LIBS:BOOL=" "" shared ${shared})
+set(BUILD_SHARED_LIBS ${shared} CACHE BOOL "" FORCE)
+message(STATUS "BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}")
+
 # Threading libraries required due to parallel pass execution.
 find_package(Threads REQUIRED)
+# libcurses and libz required by libLLVMSupport
+find_package(Curses REQUIRED)
+find_package(ZLIB REQUIRED)
 
 function(find_mlir_lib lib)
   find_library(${lib}
           NAMES ${lib}
           PATHS ${LLVM_PROJECT_LIB}
           NO_DEFAULT_PATH)
+  if(${${lib}} STREQUAL ${lib}-NOTFOUND)
+    message(FATAL_ERROR "${lib} not found")
+    #else()
```

Comment:
Seems like debug code, should we remove?

Reply:
Yes remove the commented else() part. Sorry missed that.